docs/howto/build-run.md

```diff
@@ -46,10 +46,11 @@ for starting out, just get either one working so you can play with the app.
   line](https://developer.android.com/studio/run/emulator-commandline.html).
 * When running on a physical device, if the device has Zulip installed
   from the Play Store, you may need to uninstall that version first.
-* Commands once you've set up:
+* Commands once you've set up (emulator has to be run manually before this command):
   * `react-native run-android` - build and run on an active emulator
```

Comment:
Should we replace all occurrences of `react-native run-android`?
As it will install app whose package name/id has suffix `debug` (as it is default setting in Android (studio)) and try to launch app without suffix `debug`. So if app without suffix `debug` is installed, then it will launch that else throe error after installing:
```
:app:installDebug
Installing APK 'app-debug.apk' on 'Samsung Galaxy S5 - 4.4.4 - API 19 - 1080x1920 - 4.4.4' for app:debug
Installed on 1 device.

BUILD SUCCESSFUL

Total time: 12.945 secs
Running adb -s 192.168.56.101:5555 reverse tcp:8081 tcp:8081
error: closed
Could not run adb reverse: Command failed: adb -s 192.168.56.101:5555 reverse tcp:8081 tcp:8081
Starting the app on 192.168.56.101:5555 (adb -s 192.168.56.101:5555 shell am start -n com.zulipmobile/com.zulipmobile.MainActivity)...
Starting: Intent { cmp=com.zulipmobile/.MainActivity }
Error type 3
Error: Activity class {com.zulipmobile/com.zulipmobile.MainActivity} does not exist.
Vishweshs-MacBook-Pro:zulip-mobile vishwesh$ 
```

One possible solution we can try is instead of proving suffix to `debug` build we should provide it to `release` build. Thoughts?

Reply:
Hmm. Yeah, this is a little annoying.

We can't change the release build's app ID -- that would make it a completely new app as far as people's phones who already have it installed are concerned. Also I'd rather keep the nice short name on the released version of the app anyway.

I would also like to avoid making a plain `react-native run-android` no longer work right. That'd create a difference between our app and every other RN app, so one more thing people have to learn when coming to our app and/or ignore when reading stuff about RN in general. We could do it if we had to, but preferable to avoid it.

Here's the key logic inside `react-native run-android`:
https://github.com/facebook/react-native/blob/46ffb1062/local-cli/runAndroid/runAndroid.js#L111

One idea: this is quite crude in just reading through AndroidManifest.xml and searching for a regex. What if we inserted an XML comment that was there just for this regex to find it? Like `<!-- package="com.zulipmobile.debug" -->`.

That is kind of silly, but no more silly than what the CLI is actually doing. A nice followup would be to send a PR upstream to replace that with (a) actually parsing the manifest -- see `local-cli/core/android/readManifest.js`! -- and (b) using the actual package name -- see in fact the only two call sites of `readManifest`, both in `local-cli/core/android/index.js`. Meanwhile, our workflow would be working smoothly already.

Reply:
Yup! it is working with some tweaks in the Android native code.